config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ListenPortEntityBuilder.java

```diff
@@ -101,6 +106,14 @@ Entity buildListenPortEntity(Bundle bundle, String name, ListenPort listenPort,
             Element tlsSettingsElement = document.createElement(TLS_SETTINGS);
             tlsSettingsElement.appendChild(createElementWithTextContent(document, CLIENT_AUTHENTICATION, tlsSettings.getClientAuthentication().getType()));
 
+            if (isNotEmpty(tlsSettings.getPrivateKey())) {
+                final PrivateKey privateKey = bundle.getPrivateKeys().get(tlsSettings.getPrivateKey());
```

Comment:
How are these private keys set?

Reply:
private key metadata is in json/yaml similar to all environment entities